src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6027,6 +6027,15 @@ private ScanTypeFlags ScanPossibleTypeArgumentList(
                     return result;
                 }
 
+                // Allow for any chain of errant commas in teh generic name.  like `Dictionary<,int>` or
+                // `Dictionary<int,,>` We still want to think of these as generics, just with missing type-arguments, vs
+                // some invalid tree-expression that we would otherwise form.
+                if (this.CurrentToken.Kind == SyntaxKind.CommaToken)
```

Comment:
nit: consider adding a parsing test showing the effect of the change

Reply:
Tests added.